### PR TITLE
[DOWNGRADE] grub2: 2.12-rc1 -> 2.06

### PR DESCRIPTION
**Pull request by bacchanalia**
workaround for https://github.com/NixOS/nixpkgs/issues/293038

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -8563,7 +8563,24 @@ with pkgs;
 
   grpc-client-cli = callPackage ../development/tools/misc/grpc-client-cli { };
 
-  grub2 = callPackage ../tools/misc/grub/default.nix { };
+  grub2_upstream = callPackage ../tools/misc/grub/default.nix { };
+
+  # downgrade to 2.06 from 163b243c2a1c4ce666d69635a90e4e08cf27d899
+  grub2 = assert lib.asserts.assertMsg
```
Comment (jsoo1) [inline pkgs/top-level/all-packages.nix:8569]: I would recommend calling this `grub2_06` and letting the call sites override like `grub2 = grub2_06`.

**Review by jsoo1 — APPROVED** [review]
I have one suggestion, but lgtm.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (bacchanalia) [reply pkgs/top-level/all-packages.nix:8569]: I was worried about possibly breaking something by having inconsistent grub versions, so I thought this way would potentially be safer.

Comment (jsoo1) [reply pkgs/top-level/all-packages.nix:8569]: Ok just a suggestion!

**Pull request merged**